### ISSUE TITLE
Cannot require Ruby's bundled gems

### ISSUE THREAD

**Issue by kddeisz** [bundler/bundler#7396]
Hi there - I'm attempting to use the `did_you_mean` gem (and specifically the `DidYouMean::SpellChecker` class) in my application and running into a some problems with bundler.

## Not included in Gemfile

If `did_you_mean` is not explicitly in my `Gemfile`, then it's sometimes possible for it to be used, and sometimes not. It depends on the load order. If something requires `did_you_mean` and the `DidYouMean::SpellChecker` gets loaded, then it's fine and I can use the constant, even though bundler wipes `did_you_mean` out of the load path when `clean_load_path` gets called.

This can lead to some pretty surprising behavior, as in development it was working, but once we deployed to production it wasn't. We've determined that it also has to do with where the gems are installed (if they're vendored or not), in which case the load order appears to slightly change.

The biggest problem here is that I cannot explicitly require `did_you_mean`, as it has been completely wiped out of the load path (along with every other bundled gem here: https://stdgems.org/#bundled-gems-ruby-265). 

## Included in Gemfile

If `did_you_mean` is explicitly in my `Gemfile`, at first I run into issues where it thinks I don't even have the gem:

```
$ bundle install --jobs=3 --retry=3 --deployment --path=${BUNDLE_PATH:-vendor/bundle}
$ bundle exec rails c
Could not find did_you_mean-1.3.1 in any of the sources
Run `bundle install` to install missing gems.
```

If I run anything with `bundle exec`, it tells me that. However, if I run it with `bin/rails`, I can get into the console and it's loaded. Except in certain environments where I run into issues where `did_you_mean` is already loaded, as in:

```
$ bundle exec rails db:create
/home/travis/.rvm/gems/ruby-2.6.3/gems/bundler-2.0.2/lib/bundler/runtime.rb:319:in `check_for_activated_spec!': You have already activated did_you_mean 1.3.0, but your Gemfile requires did_you_mean 1.3.1. Prepending `bundle exec` to your command may solve this. (Gem::LoadError)
```

which appears to happen if certain things are required before others and `did_you_mean` gets activated by something.

## Future work

I'm really not sure of the best answer here. At the moment I'm unable to get this functionality working because bundler is wiping `did_you_mean` out of the load path if it's not in the Gemfile, and in certain circumstances is unable to load it correctly if it is in the Gemfile. 

I can see this has been discussed before here: https://github.com/bundler/bundler/issues/6648. It seems like from that discussion that the bundler team wants you to explicitly list Ruby's bundled gems in your Gemfile.

Either way it seems like this really needs to be in the docs, I scoured them looking for information on bundled gems and couldn't find anything on them. As more of the stdlib gets gemified and decisions get made about which gems become "bundled" versus "standard", this is going to become more of a problem for people that have explicit requires because they want that functionality.

If someone has any ideas or can point me in the right direction, I'd really appreciate it. I'd be more than willing to help if there's something actionable here.

**Comment by hsbt** [comment]
It's related with https://github.com/yuki24/did_you_mean/issues/117#issuecomment-482733159

**Comment by kddeisz** [comment]
I'm going to close this as it's related to `did_you_mean` becoming a default gem. It'll be fixed incidentally by ruby 2.7, so not going to worry about it here.
